ivadomed/scripts/extract_small_dataset.py

```diff
@@ -71,21 +71,21 @@ def extract_small_dataset(input, output, n=10, contrast_list=None, include_deriv
             each function run is independent. Flag: ``--seed``, ``-s``.
     """
     # Create output folders
-    if not os.path.isdir(output):
-        os.makedirs(output)
+    if not Path(output).is_dir():
+        Path(output).mkdir(parents=True)
     if include_derivatives:
-        oderivatives = os.path.join(output, "derivatives")
-        if not os.path.isdir(oderivatives):
-            os.makedirs(oderivatives)
-        oderivatives = os.path.join(oderivatives, "labels")
-        if not os.path.isdir(oderivatives):
-            os.makedirs(oderivatives)
-        iderivatives = os.path.join(input, "derivatives", "labels")
+        oderivatives = Path(output, "derivatives")
+        if not oderivatives.is_dir():
+            oderivatives.mkdir(parents=True)
+        oderivatives = Path(oderivatives, "labels")
+        if not oderivatives.is_dir():
+            oderivatives.mkdir(parents=True)
+        iderivatives = Path(input, "derivatives", "labels")
 
     # Get subject list
-    subject_list = [s for s in os.listdir(input)
-                    if s.startswith("sub-") and os.path.isdir(os.path.join(input, s))
-                    and s not in EXCLUDED_SUBJECT]
+    subject_list = [s.name for s in Path(input).iterdir()
+                    if s.name.startswith("sub-") and s.is_dir()
+                    and s.name not in EXCLUDED_SUBJECT]
```

Comment:
MUCH more readable and easier to follow than original. While I am not fan of multi-line list comprehensions, this is clear enough so we can leave it alone.